lib/pbench/server/indexer.py

```diff
@@ -3603,9 +3604,11 @@ def __init__(self, idxctx, tbarg, tmpdir, extracted_root):
                     'run.controller ("{}") does not match'
                     ' controller_dir ("{}")'.format(controller, self.controller_dir)
                 )
+            self.controller = controller
```

Comment:
Perhaps add a comment that this is a convenience field since the `controller` value is also available as `self.run_metadata["controller"]`?

Also, since on-disk we use `controller_dir`, should we be using that value?

Reply:
> Also, since on-disk we use `controller_dir`, should we be using that value?

My intent was really just to come up with a minimal set of unique metadata that would universally identify a particular dataset. I came up with user/controller/name. The concept of `controller_dir` may actually go away entirely (or become awkward and arbitrary) when we move to S3 storage instead of an on-disk hierarchy.

Metadata is the way to go, and I think "controller" is arguably more "real" metadata than "controller_dir". Do you disagree?